pkg/apis/pipeline/v1alpha1/verificationpolicy_validation.go

```diff
@@ -61,8 +61,16 @@ func (vs *VerificationPolicySpec) Validate(ctx context.Context) (errs *apis.Fiel
 // Validate KeyRef will check if one of KeyRef's Data or SecretRef exists, and the
 // Supported HashAlgorithm is in supportedSignatureAlgorithms.
 func (key *KeyRef) Validate(ctx context.Context) (errs *apis.FieldError) {
-	if key.Data == "" && key.SecretRef == nil {
-		errs = errs.Also(apis.ErrMissingOneOf("data", "secretref"))
+	if key.Data == "" && key.SecretRef == nil && key.KMS == "" {
+		errs = errs.Also(apis.ErrMissingOneOf("data", "kms", "secretref"))
+	}
+
+	if key.Data != "" {
+		if key.KMS != "" || key.SecretRef != nil {
+			errs = errs.Also(apis.ErrMultipleOneOf("data", "kms", "secretref"))
+		}
+	} else if key.KMS != "" && key.SecretRef != nil {
+		errs = errs.Also(apis.ErrMultipleOneOf("data", "kms", "secretref"))
 	}
 
 	if key.Data != "" && key.SecretRef != nil {
```

Comment:
nit:
This entire piece can be recoded as 
num_key_types := 0
if key.Data != "" {
   num_key_types += 1
}
if key.SecretRef != nil {
   num_key_types +=1 
}
..
if num_key_types == 0 {
   errs = errs.Also(apis.ErrMissingOneOf("data", "secretref"))
}
if num_key_types > 1 {
   errs = errs.Also(apis.ErrMultipleOneOf("data", "kms", "secretref"))
}

Which would be more concise and clear.

Reply:
thank! I agree this has more readability, I will change the code!